meiga/result.py

```diff
@@ -135,11 +124,11 @@ def unwrap_or_else(
                         on_failure()
                     else:
                         on_failure(self.value)
-            return failure_value
+            return cast(TEF, failure_value)
         return cast(TS, self.value)
 
     def unwrap_and(
-        self, on_success: Callable, success_args: Optional[Any] = None
+        self, on_success: Callable[..., None], success_args: Optional[Any] = None
```

Comment:
A first approximation has been added to better type the derived functions (`Callable[..., None]`). 
meiga expects the  side-effect methods not to return any values

On the other hand, we've tried some alternatives to link the Callable input parameters to those of the arguments that we pass in (e.g `success_args`)


```python
from typing import ParamSpec

PS = ParamSpec("PS")

def unwrap_and(self, on_success: Callable[PS, None], success_args: Optional[PS] = None)
```

unfortunately, we have not yet found a solution in this regard.